python-sdk/src/langwatch/telemetry/__tests__/test_tracing.py

```
@@ -0,0 +1,48 @@
+
+
+def test_metadata_on_root_span():
+    # Create trace with metadata and root span
+    trace = LangWatchTrace(metadata={"a": 1, "b": 2})
+    with trace:
+        trace.update(metadata={"b": 3, "c": 4})
+        # The root span's metadata attribute should match the merged metadata
+        root_metadata = trace.root_span._span.attributes.get("metadata")
+        import json
+
+        assert json.loads(root_metadata) == {"a": 1, "b": 3, "c": 4}
```

Ruff (0.12.2)

Use of `assert` detected

(S101)